src/storage/heap_file.c

```diff
@@ -7279,17 +7283,34 @@ heap_prepare_get_context (THREAD_ENTRY * thread_p, HEAP_GET_CONTEXT * context,
 	  if (bcb_area_copied)
 	    {
 	      page_ptr = context->bcb_area;
+	      perfmon_inc_stat (thread_p, PSTAT_HEAP_NUM_COPY_NOLATCH_SUCCESS);
+	      pgbuf_copy_log ("pgbuf_copy_page: Successfully copied heap page (%d, %d) \n",
+			      object_vpid.volid, object_vpid.pageid);
 	    }
 	  else
 	    {
+	      perfmon_inc_stat (thread_p, PSTAT_HEAP_NUM_COPY_NOLATCH_FAILED);
```

Comment:
please introduce a metric to count the number of retries and separate "retry" and "failure".